tests/test_configurations.py

```diff
@@ -25,14 +30,47 @@ def can_i_run_smrf(self, config):
             # print(e)
             return False
 
+    def compare_netcdf_files(self, gold_file, test_file):
+        """
+        Compare two netcdf files to ensure that they are identical. The
+        tests will compare the attributes of each variable and ensure that
+        the values are exact
+        """
+
+        # open the gold and compare
+        gold = nc.Dataset(gold_file)
+        test = nc.Dataset(test_file)
+
+        # compare the time first
+        self.assertEqual(len(gold.variables['time']), len(
+            test.variables['time']))
+
+        # go through all variables and compare everything including the attributes and data
+        for var_name, v in gold.variables.items():
+
+            # compare the dimensions
+            for att in gold.variables[var_name].ncattrs():
+                self.assertEqual(
+                    getattr(gold.variables[var_name], att),
+                    getattr(test.variables[var_name], att))
+
+            # only compare those that are floats
+            if gold.variables[var_name].datatype != np.dtype('S1'):
+                result = np.abs(
+                    gold.variables[var_name][:] - test.variables[var_name][:])
+                self.assertTrue(not np.any(result > 0))
+
+        gold.close()
+        test.close()
```

Comment:
Could wrap lines 44 to 61 in a `try` and `finally` block to ensure the files are closed when one of the assertions are failing.

Reply:
From my understanding, it will not raise an exception here but log that it failed. So it should always make it to the close statements. Your method would be a lot more explicit